### ISSUE TITLE
[SGS2] Stays on Starting... forever and plays microphone

### ISSUE THREAD

**Issue by dllud** [mikereidis/spirit2_free#5]
When I open SpiritF it stays on the "SpiritF Starting..." screen forever. If I touch somewhere else on the screen it dismisses the starting... message but continues without playing. When press stop (square button) followed by play/pause repeatedly it sometimes starts and shows me the frequency value on the screen. However, on the headphones what I hear is my environment through the phone's microphone. Still, most of the times, when I press the stop/play/pause buttons repeatedly, the phone crashes and reboots.

I could spot nothing of interest on logcat. Though, if you pass me an adequate filter (there is too much output), maybe I can post a useful debug log here.

This happens on a Samsung Galaxy S II (GT-i9100) running Replicant 4.2 (based on CM10.1, Android 4.2.2) with the lastest apk from F-Droid (2015_01_17_beta).

Spirit1 Light works like a charm here.

**Comment by mikereidis** [comment]
Replicant 4.2 you say ? :)

The Lollipop re-design of Spirit2 just doesn't work well on some
pre-Android 5.0 ROMs.

Try the Kitkat release that I put out for ICS-Kitkat after too
many reports. ​

**Comment by dllud** [comment]
You are right. The Kitkat release works here (with a bug on the speaker output which works for no more than 1 second).

Was this Kitkat release built from the same code you have here on the repo? If so, couldn't you put both builds into F-Droid, while adjusting minSdkVersion in the AndroidManifest accordingly?

**Comment by aelmahmoudy** [comment]
Where is that Kitkat release ? I only see on branch on git

**Comment by mikereidis** [comment]
Spirit2 was substantially redesigned for Lollipop. Took 3+ months of long hours to do.

The Kitkat (Spirit2 Classic) release was a "one-off" that I only did to get some of the pre-Lollipop users up and running again.

Going forward, Spirit2 Classic is frozen and will likely not receive any updates/bug fixes. Problems will only be fixed in the redesigned code. Just too much work otherwise and same for releasing it's source (which will also confuse people and end up requiring hours of answering questions.)

Kitkat release 141128 is based on the Spirit2 140912 September 12 release code, which had some speaker issues on some ROMs like Omni. September 11 worked better on those and you can email me for a copy.

**Comment by dllud** [comment]
Ok. As this code only supports post-Android 5.0 ROMs I think you should adjust minSdkVersion in the AndroidManifest accordingly and then close this issue.

**Comment by mikereidis** [comment]
No, this code supports Android 4.1 and up. It would be 4.0+ except that the
binaries are PIE.

It just happens to have problems on a FEW ROMs, regardless of Android
version.

So let's see if we can fix the problems on whatever ROMs have problems.

The new 150127 release today has a LOT of stability and efficiency
improvements.

Perhaps some of them help. Please test.

Post a clear description of the problems, the ROM it affects, and where I
can get the ROM and source code.

Then send me or post the appropriate kernel, logcat, bugreport logs etc.

You can send me a bugreport style debug log from Spirit2/Free. Tap the
frequency button, enter the number 7 and tap "OK". After 20-40 seconds or
so, press Send from your email client.

**Comment by aelmahmoudy** [comment]
Yes, it works on OMNI 4.4.4 ! Thanks a lot !

**Comment by mikereidis** [comment]
Good. :)

Maybe I should add that the next source release here will likely only support Android 4.4 Kitkat and up.

I ran lint, and noticed that I was using APIs restricted to 4.4+ and this may have been causing crashes in 4.1-4.3.

The quickest, simplest fix at this time is just to restrict to 4.4+. I only test on 4.4+ and don't expect that to change.

**Comment by dllud** [comment]
> The quickest, simplest fix at this time is just to restrict to 4.4+.

I agree, although this means I won't be able to use Spirit2 Free...  Go for it.
